benchmarks/test_benchmarks.py

```diff
@@ -0,0 +1,132 @@
+import functools as ft
+import warnings
+
+import jax
+import optimistix as optx
+import pytest
+import scipy as scp
+import sif2jax
+
+from .conftest import get_max_dimension
+
+
+cutest = pytest.mark.skipif("not config.getoption('cutest')")
+scipy = pytest.mark.skipif("not config.getoption('scipy')")
+
+
+# Set a consistent number of maximum steps for all solvers. We might want to override
+# this, make it solver-specific in the future, or make it a config option.
+max_steps = 2**8
+
+# Benchmark solvers that are part of documented API.
+unconstrained_minimisers = (optx.BFGS(rtol=1e-3, atol=1e-6),)
+# Specify scipy minimisers using tuples (method_name: str, kwargs: dict)
+unconstrained_scipy_minimisers = (("BFGS", {}),)  # TODO(jhaffner): scipy tolerances?
+
+
+def get_test_cases(problems):
+    max_dimension = get_max_dimension()
+    if max_dimension is not None:
+        test_cases = []
+        for problem in problems:
+            problem_dimension = problem.y0().size
+            if problem_dimension <= max_dimension:
+                test_cases.append(problem)
+        return tuple(test_cases)
+    else:
+        return problems
+
+
+@cutest
+@pytest.mark.benchmark
+@pytest.mark.parametrize(
+    "problem", get_test_cases(sif2jax.unconstrained_minimisation_problems)
+)
+@pytest.mark.parametrize("minimiser", unconstrained_minimisers)
+def test_runtime_unconstrained_minimisers(benchmark, minimiser, problem):
+    solve = jax.jit(
+        ft.partial(
+            optx.minimise,
+            problem.objective,
+            minimiser,
+            args=problem.args(),
+            max_steps=max_steps,
+            throw=False,
+        )
+    )
+
+    _ = solve(problem.y0())  # Warm up
+
+    def wrapped(y0):
+        solution = solve(y0)
+        objective_value = solution.state.f_info.f.block_until_ready()
+        num_steps = solution.stats["num_steps"]
+        return objective_value, solution.result, num_steps
+
+    # Benchmark the runtime of the compiled function
+    values = benchmark.pedantic(wrapped, args=(problem.y0(),), rounds=5, iterations=1)
```

Comment:
How does the benchmarking work here btw? As compared to e.g. `min(timeit.repeat(..., number=..., repeat=...))` or the ipython `%%timeit`? (E.g. the former uses minimisation to handle one-sided noise, whilst the latter automatically figures out how many runs to perform.)